exchanges/bybit/bybit_test.go

```diff
@@ -3111,62 +3111,145 @@ func TestPushData(t *testing.T) {
+	}
 	sharedtestvalues.TestFixtureToDataHandler(t, b, n, "testdata/wsTicker.json", func(r []byte) error {
 		defer slices.Delete(assetRouting, 0, 1)
 		return n.wsHandleData(assetRouting[0], r)
 	})
 	close(n.Websocket.DataHandler)
-	assert.Len(t, n.Websocket.DataHandler, 4, "Should see 4 tickers distributed")
+	expected := 8
+	require.Len(t, n.Websocket.DataHandler, expected, "Should see correct number of tickers")
 	for resp := range n.Websocket.DataHandler {
 		switch v := resp.(type) {
 		case *ticker.Price:
 			assert.Equal(t, n.Name, v.ExchangeName, "ExchangeName should be correct")
-			switch len(n.Websocket.DataHandler) {
-			case 3:
+			switch expected - len(n.Websocket.DataHandler) {
+			case 1: // Spot
+				assert.Equal(t, currency.BTC, v.Pair.Base, "Pair base should be correct")
+				assert.Equal(t, currency.USDT, v.Pair.Quote, "Pair quote should be correct")
 				assert.Equal(t, 21109.77, v.Last, "Last should be correct")
 				assert.Equal(t, 21426.99, v.High, "High should be correct")
 				assert.Equal(t, 20575.00, v.Low, "Low should be correct")
 				assert.Equal(t, 6780.866843, v.Volume, "Volume should be correct")
 				assert.Equal(t, "BTC_USDT", v.Pair.String(), "Pair should be correct")
 				assert.Equal(t, asset.Spot, v.AssetType, "AssetType should be correct")
-				assert.EqualValues(t, 233366401, v.LastUpdated.Unix(), "LastUpdated should be correct")
-			case 2:
-				assert.Equal(t, 17216.00, v.Last, "Last should be correct")
-				assert.Equal(t, 17281.50, v.High, "High should be correct")
-				assert.Equal(t, 16915.00, v.Low, "Low should be correct")
-				assert.Equal(t, 91705.276, v.Volume, "Volume should be correct")
-				assert.Equal(t, 17215.50, v.Bid, "Bid should be correct")
-				assert.Equal(t, 84.489, v.BidSize, "BidSize should be correct")
-				assert.Equal(t, 17216.00, v.Ask, "Ask should be correct")
-				assert.Equal(t, 83.020, v.AskSize, "AskSize should be correct")
-				assert.Equal(t, "BTC_USDT", v.Pair.String(), "Pair should be correct")
-				assert.Equal(t, asset.USDTMarginedFutures, v.AssetType, "AssetType should be correct")
-				assert.EqualValues(t, 233366402, v.LastUpdated.Unix(), "LastUpdated should be correct")
-			case 1:
-				assert.Equal(t, 10.00, v.Last, "Last should be correct")
-				assert.Equal(t, 25.0, v.High, "High should be correct")
-				assert.Equal(t, 5.00, v.Low, "Low should be correct")
-				assert.Equal(t, 26.55, v.Volume, "Volume should be correct")
-				assert.Equal(t, 7.00, v.Bid, "Bid should be correct")
-				assert.Equal(t, 4.0, v.BidSize, "BidSize should be correct")
-				assert.Equal(t, 11.00, v.Ask, "Ask should be correct")
-				assert.Equal(t, 5.10, v.AskSize, "AskSize should be correct")
-				assert.Equal(t, "BTC-USD-220930-28000-P", v.Pair.String(), "Pair should be correct")
+				assert.EqualValues(t, 233366401000, v.LastUpdated.UnixMilli(), "LastUpdated should be correct")
```

Comment:
looks like you'll need to wrap with int64(...)

Reply:
Yep. That's EqualValues out the window for most use cases then 🪟 

Fixed 2639b119